src/libmongoc/src/mongoc/mongoc-cluster.c

```diff
@@ -677,13 +678,16 @@ mongoc_cluster_run_command_private (mongoc_cluster_t *cluster,
    if (!reply) {
       reply = &reply_local;
    }
+
    server_stream = cmd->server_stream;
-   if (server_stream->sd->max_wire_version >= WIRE_VERSION_OP_MSG) {
+
```

Comment:
`mongoc_cluster_t` contains a pointer to `mongoc_client_t`. Instead of having a `force_op_msg` bool as a parameter, consider:

```c
bool force_op_msg = mongoc_client_uses_server_api (cluster->client);
```

Reply:
I like that function name-- and I agree that it's clear now that parameter doesn't need to come from a distance. Once tests are passing I'll implement this (and your related suggestion to consolidate the various tests for the client having specified a server API).

Reply:
I've added a few functions (in the style of overloads) to do this, as sometimes the right path through the nested structs was tricky to find.